tt-rss: 2021-01-29 -> 2021-06-23 and modules/tt-rss: updated config.php creation

milogert opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

The version of tt-rss in nixpkgs was pre-php 8. This updates tt-rss to the most recent version, which uses php 8.

The updates to the module are required changes, per https://community.tt-rss.org/t/rip-config-php-hello-classes-config-php/4337.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

milogert committed:
tt-rss: 2021-01-29 -> 2021-06-23 and modules/tt-rss: updated config.php creation

The version of tt-rss in nixpkgs was pre-php 8. This updates tt-rss to the most recent version, which uses php 8.

The updates to the module are required changes, per https://community.tt-rss.org/t/rip-config-php-hello-classes-config-php/4337.

milogert committed:
tt-rss: 2021-01-29 -> 2021-06-23 and modules/tt-rss: updated config.php creation

The version of tt-rss in nixpkgs was pre-php 8. This updates tt-rss to the most recent version, which uses php 8.

The updates to the module are required changes, per https://community.tt-rss.org/t/rip-config-php-hello-classes-config-php/4337.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 128141` at bdb0bbef run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>tt-rss</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/servers/tt-rss/default.nix:13:3:
```
   |
13 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

symphorien commented:
I can review this seriously in about a week.

Maybe you will be interested in #122095 ?

milogert committed:
Api didn't seem to work with the systemd envs

I am not sure why it was working for me, but after restarting
phpfpm-tt-rss it quit working.

milogert committed:
Api didn't seem to work with the systemd envs

I am not sure why it was working for me, but after restarting
phpfpm-tt-rss it quit working.

symphorien commented:
I get this on upgrade:
```
Exception while creating PDO object:SQLSTATE[08006] [7] FATAL:  database "host=db" does not exist
```

symphorien commented:
Ah I had to restart php-fpm. Could you add the config to https://search.nixos.org/options?channel=21.05&show=systemd.services.%3Cname%3E.restartTriggers&from=0&size=50&sort=relevance&query=systemd+service+restart for the php fpm service?

symphorien commented:
This also needs release notes to tell users of `extraConfig` that they need to migrate from `define` to `putenv` I suspect.

milogert commented:
@symphorien I am happy to add release notes. How do I go about doing that?

Also, thanks for the suggestion on the `restartTriggers`

symphorien commented:
https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes

symphorien commented:
Can you also add 
```diff
diff --git a/pkgs/servers/tt-rss/default.nix b/pkgs/servers/tt-rss/default.nix
index bd82f8d2890..61ec3eca824 100644
--- a/pkgs/servers/tt-rss/default.nix
+++ b/pkgs/servers/tt-rss/default.nix
@@ -2,7 +2,11 @@
 
 stdenv.mkDerivation rec {
   pname = "tt-rss";
-  version = "2021-06-21";
+  /* date of the commit ${rev} */
+  year = "21";
+  month = "06";
+  day = "21";
+  version = "20${year}-${month}-${day}";
   rev = "cd26dbe64c9b14418f0b2d826a38a35c6bf8a270";
 
   src = fetchurl {
@@ -13,6 +17,8 @@ stdenv.mkDerivation rec {
   installPhase = ''
     mkdir $out
     cp -ra * $out/
+    # see the code of Config::get_version(). you can check that the version in the footer of the preferences pages is not UNKNOWN
+    echo "${year}.${month}" > $out/version_static.txt
   '';
 
   meta = with lib; {

```
otherwise tt-rss reports the wrong version to plugins, which then may fail.

It's not directly related to your PR, if you don't want to include that I'll make a PR later.

symphorien commented:
ping @milogert

milogert committed:
Added release notes and broke date apart

milogert committed:
Merge branch 'tt-rss-2021-06-21' of github.com:milogert/nixpkgs into tt-rss-2021-06-21

milogert commented:
@symphorien Sorry! I had some life stuff come up (moving across country). I made the changes you requested, but I merged in master, so I think it accidentally over-requested reviews from a bunch of people :grimacing:

milogert committed:
Merge branch 'master' of https://github.com/NixOS/nixpkgs into tt-rss-2021-06-21

SuperSandro2000 commented on nixos/modules/services/web-apps/tt-rss.nix:
```diff
@@ -604,6 +611,9 @@ let
           ''}
           ln -sf "${tt-rss-config}" "${cfg.root}/config.php"
           chmod -R 755 "${cfg.root}"
+          chmod -R 777 "${cfg.root}/${lockDir}"
+          chmod -R 777 "${cfg.root}/${cacheDir}"
+          chmod -R 777 "${cfg.root}/${feedIconsDir}"
```
No, 777 is bad.

symphorien replied:
you can use `ug+rwX` to only set the executable bit on directories I think.

symphorien commented on nixos/doc/manual/from_md/release-notes/rl-2111.section.xml:
```diff
@@ -555,6 +555,14 @@
+        <para>
+          <literal>tt-rss</literal> was upgraded to the commit on
+          2021-06-21, which has breaking changes, see
+          <link xlink:href="https://community.tt-rss.org/t/rip-config-php-hello-classes-config-php/4337">this
```
It would be nice to be more specific about what adaptations must be done. I'm thinking about something like "If you use `services.ttrss.extraConfig` you should migrate to `putenv` style configuration. See this discourse post for details."

symphorien commented on nixos/modules/services/web-apps/tt-rss.nix:
```diff
@@ -564,9 +566,14 @@ let
+    systemd.services = {
+      phpfpm-tt-rss = mkIf (cfg.pool == "${poolName}") {
+        restartTriggers = [
+          cfg.root
```
`restartTriggers` only works with store paths. (In reality, restart triggers only trigger a restart when their stringified constant changes). 

So I suggest `restartTriggers  = [ tt-rss-config tt-rss ];`

symphorien commented:
Thanks!
I think we are getting close, I'm going to deploy this to test it.

milogert committed:
Added more detail to changelog, updated permissions in directory, and changed restartTriggers

milogert committed:
Merge branch 'master' into tt-rss-2021-06-21

symphorien commented on nixos/modules/services/web-apps/tt-rss.nix:
```diff
@@ -564,9 +566,12 @@ let
-      {
+    systemd.services = {
+      phpfpm-tt-rss = mkIf (cfg.pool == "${poolName}") {
+        restartTriggers = [ tt-rss-config tt-rss ];
```
```suggestion
        restartTriggers = [ tt-rss-config pkgs.tt-rss ];
```
sorry for the misleading review

symphorien committed:
nixos/tt-rss: fix eval

symphorien commented:
Thanks for your work!

symphorien commented:
if you did an upgrade, maybe you need to run a schema upgrade ?
is `systemctl status tt-rss.service` happy?
